tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,530 @@
+"""
+
+import json
+from typing import Dict
+
+import pytest
+import requests
+from flask.testing import FlaskClient
+from synapseclient.client import Synapse
+
+from tests.conftest import ConfigurationForTesting, Helpers
+
+EXAMPLE_SCHEMA_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+
+
+@pytest.fixture
+def request_headers(syn_token: str) -> Dict[str, str]:
+    """Simple bearer token header for requests"""
+    headers = {"Authorization": "Bearer " + syn_token}
+    return headers
+
+
+class TestManifestValidation:
+    @pytest.mark.local_or_remote_api
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            ("Biospecimen", "mock_manifests/example_biospecimen_test.csv"),
+            ("Patient", "mock_manifests/example.patient_component_rule.manifest.csv"),
+        ],
+    )
+    def test_manifest_validation_basic_valid(
+        self,
+        input_data_type: str,
+        input_file_name: str,
+        flask_client: FlaskClient,
+        request_headers: Dict[str, str],
+        testing_config: ConfigurationForTesting,
+        helpers: Helpers,
+    ) -> None:
+        """
+        Test that the manifest validation API returns no errors when a valid manifest is provided.
+        """
+        # GIVEN the manifest validation endpoint and parameters
+        url = f"{testing_config.schematic_api_server_url}/v1/model/validate"
+        params = {
+            "schema_url": EXAMPLE_SCHEMA_URL,
+            "data_type": input_data_type,
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+        }
+
+        # AND a valid file
+        file_path = helpers.get_data_path(input_file_name)
+
+        # WHEN we make a POST request to validate the file
+        response = (
+            requests.post(
+                url,
+                params=params,
+                files={"file_name": open(file_path, "rb")},
+                headers=request_headers,
+            )
+            if testing_config.use_deployed_schematic_api_server
+            else flask_client.post(
+                url,
+                headers=request_headers,
+                query_string=params,
+                data={"file_name": open(file_path, "rb")},
+            )
+        )
+
+        # THEN we expect a successful response
+        assert (
+            response.status_code == 200
+        ), f"Got status code: {response.status_code}. Expected '200'."
+
+        # AND with expected keys in the json
+        response_json = (
+            response.json()
+            if testing_config.use_deployed_schematic_api_server
+            else response.json
+        )
+        assert "warnings" in response_json.keys()
+        assert "errors" in response_json.keys()
+
+        # AND with no expected errors
+        assert len(response_json.get("errors")) == 0
```

Comment:
Could we also assert that there are no warnings?

Reply:
Added! https://github.com/Sage-Bionetworks/schematic/pull/1516/commits/0828e340c65709d8bee4eaebe05d1af44de2a72a